### PR TITLE
Maintenance/576/migrate to GitHub actions

### PR DESCRIPTION
**Pull request by cliffle**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+name: Continuous integration
+
+on:
+  push:
+    branches-ignore:
+      - "master"
+      - "develop"
```
Comment (svettwer) [inline .github/workflows/ci.yml:7]:
```suggestion
    branches-ignore:
      - "master"
      - "develop"
      - "release/**"
```

Other than in all other projects, release branches are not ignored by the ci pipeline. Is this intended? If yes: Should we discuss to run ci build on release branches as well?

File: .github/workflows/cd.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      - name: Run unit/integration tests
+        uses: GabrielBB/xvfb-action@v1
+        with:
+          run: "bash run_${{ runner.os }}.sh"
+      - name: Send results to SonarCloud
+        if: runner.os == 'Linux'
+        uses: SonarSource/sonarcloud-github-action@v1.4
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
```
Comment (svettwer) [inline .github/workflows/cd.yml:33]:
```suggestion
```
I think it is not required to send coverage information on cd builds. At the end, cd just makes sure that a new next release is created from the merged pr. There should not be any code change on develop.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      - name: Run unit/integration tests
+        uses: GabrielBB/xvfb-action@v1
+        with:
+          run: "bash run_${{ runner.os }}.sh"
+      - name: Send results to SonarCloud
+        if: runner.os == 'Linux'
+        uses: SonarSource/sonarcloud-github-action@v1.4
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
```
Comment (svettwer) [inline .github/workflows/release.yml:35]:
```suggestion
```
same as for cd. I think we can omit QA analysis on release builds as no code change will happen on master or tags.

**Review by svettwer — CHANGES_REQUESTED** [review]

**Comment by svettwer** [comment]
Please remove `Travis CI - Pull Request` from required checks in branch config as well.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (cliffle) [reply .github/workflows/ci.yml:7]: There was no specific reason. I copied this from `sakuli-dashboard`. I think it's the only one where the release branch is not ignored :see_no_evil:

**Review by svettwer — APPROVED** [review]

**Pull request merged**